dev_apps/music-nga/elements/music-view-stack.js

```diff
@@ -77,132 +87,161 @@ proto.createdCallback = function() {
   this.innerHTML = template;
 
   this.container = this.querySelector('.music-view-stack-container');
-  this.container.addEventListener('transitionend', (evt) => {
+  this.container.addEventListener('animationend', (evt) => {
     var classList = evt.target.classList;
-    if (classList.contains('pop') &&
-        classList.contains('out')) {
-      evt.target.contentWindow.dispatchEvent(new CustomEvent('viewdestroy'));
+    if (classList.contains('destroy')) {
+      this.destroyView(evt.target.src);
     }
 
     classList.remove('fade');
-    classList.remove('push');
     classList.remove('pop');
+    classList.remove('push');
     classList.remove('in');
     classList.remove('out');
-    classList.remove('transition');
```

Comment:
You can put these all in one statement, like `classList.remove('fade', 'pop', 'push', ...);`